packages/webdriverio/src/middlewares.js

```diff
@@ -32,7 +33,19 @@ export const elementErrorHandler = (fn) => (commandName, commandFn) => {
                     () => {
                         return this.parent.$(this.selector).then((elem) => {
                             this.elementId = elem.elementId
-                            return fn(commandName, commandFn).apply(this, args)
+                            try{
+                                return fn(commandName, commandFn).apply(this, args)
+                            } catch(error) {
+                                if (error.message.includes("stale element reference")) {
+                                    refetchElement(this).then(element => {
```

Comment:
still missing a `return`

Reply:
once the `refetchElement` method finished we can continue with the original `fn(commandName, commandFn)` but we definitely wait until all element were refetched

Reply:
Why does this need a return? Theres a .then() off of the returned value of the function that then updates the parent and element id of the this context. We then return the function call itself with the updated context.

Reply:
I think the problem is indentation on the return...

Reply:
Done. I tried the async path, but it just turned into a rabbit hole.